src/codewhisperer/service/inlineCompletionService.ts

```diff
@@ -90,14 +90,15 @@ class CWInlineCompletionItemProvider implements vscode.InlineCompletionItemProvi
     truncateOverlapWithRightContext(document: vscode.TextDocument, suggestion: string): string {
         let rightContextRange: vscode.Range | undefined = undefined
         const pos = vscode.window.activeTextEditor?.selection.active || RecommendationHandler.instance.startPos
-        if (suggestion.split(/\r?\n/).length > 1) {
-            rightContextRange = new vscode.Range(pos, document.positionAt(document.offsetAt(pos) + suggestion.length))
+        const lines = suggestion.split(/\r?\n/).length
+        if (lines > 1) {
+            rightContextRange = new vscode.Range(pos, document.lineAt(pos.line + lines).range.end)
         } else {
-            rightContextRange = new vscode.Range(pos, document.lineAt(pos).range.end)
+            rightContextRange = new vscode.Range(pos, document.lineAt(pos.line + 1).range.end)
```

Comment:
Could you explain why this + 1 is necessary? Thanks!

Reply:
this is for when I invoke at line1, suggestion is one line and an exact match of my right context at line 2